src/lib/operators/jit_operator/operators/jit_expression.hpp

```diff
@@ -29,22 +54,36 @@ class JitExpression {
   std::shared_ptr<const JitExpression> right_child() const { return _right_child; }
   const JitTupleValue& result() const { return _result_value; }
 
-  /* Triggers the (recursive) computation of the value represented by this expression.
-   * The result is not returned, but stored in the _result_value tuple value.
-   * The compute() function MUST be called before the result value in the runtime tuple can safely be accessed through
-   * the _result_value.
+  // The compute() and compute<ResultValueType>() functions trigger the (recursive) computation of the value
+  // represented by this expression.
+
+  /* The compute() function does not return the result, but stores it in the _result_value tuple value.
```

Comment:
Maybe rename to `compute_and_store`? I don't like having two functions with the same name that behave fundamentally different and are only differentiated by their template argument.